contracts/strategies/LPOptimiserStrategy.sol

```diff
@@ -43,11 +57,133 @@ contract LPOptimiserStrategy is DefaultAccessControl, ILpCallback {
+        _requireAtLeastOperator();
+        _logProximity = logProx;
+        _sigmaWad = sigmaWad;
+        _max_possible_lower_bound = max_possible_lower_bound;
+        _tickSpacing = tickSpacing;
+    }
+
+    function setTickValues(int24 currentTick, int24 tickLower, int24 tickUpper) public {
+        _requireAtLeastOperator();
+        _currentTick = currentTick;
+        _tickLower = tickLower;
+        _tickUpper = tickUpper;
+    }
+
+    /// @notice Get the current tick and position ticks and decide whether to rebalance
+    function rebalanceCheck() public view returns (bool) {
+        
+        // 1. Get current position, lower, and upper ticks
+        // _currentPosition = _vault.currentPosition();
+        // int24 _tickLower = _currentPosition.tickLower;
+        // int24 _tickUpper = _currentPosition.tickUpper;
+
+        // 2. Get current tick
+        // int24 _currentTick = _periphery.getCurrentTick(_periphery.address());
+
+        // 3. Compare current fixed rate to lower and upper bounds
+        if (
+            _tickLower - _logProximity <= _currentTick &&
+            _currentTick <= _tickUpper + _logProximity
+            
+            ) {
+            // 4.1. If current fixed rate is within bounds, return false (don't rebalance)
+            return (false);
+        } else {
+            // 4.2. If current fixed rate is outside bounds, return true (do rebalance)
+            return (true);
+        }
+    }
+
+    /// @notice Get the nearest tick multiple given a tick and tick spacing
+    function nearestTickMultiple(int24 newTick, int24 tickSpacing) internal pure returns (int24) {
+     return (newTick / tickSpacing + (newTick % tickSpacing >= tickSpacing/2 ? int24(1) : int24(0)) ) * tickSpacing;
+    }
+
+    /// @notice Set new optimimal tick range based on current tick
+    /// @param currentFixedRateWad currentFixedRate which is passed in from a 7-day rolling avg. historical fixed rate.
+    function rebalance (int256 currentFixedRateWad) public returns (int256 newTickLower, int256 newTickUpper) {
+        _requireAtLeastOperator();
+
+        // console.log('my console logs start here:');
+        // console.logInt(PRBMathSD59x18.log2(currentFixedRateWad));
+        // console.logInt(PRBMathSD59x18.log2(1000100000000000000));
+        // console.logInt(-PRBMathSD59x18.div(PRBMathSD59x18.log2(currentFixedRateWad), PRBMathSD59x18.log2(1000100000000000000)));
+        // console.log(PRBMathUD60x18.mul(_sigmaWad, 2000000000000000000));
+        // console.logInt(nearestTickMultiple(_newTickLower, _tickSpacing));
+
+        if (rebalanceCheck()) {
+            // 0. Get tickspacing from vamm
+            // int24 _tickSpacing = _vamm.tickSpacing(_vamm.address());
+
+            // 1. Get the new tick lower
+            // uint256 _newFixedLowerWad = Math.min(Math.max(0, uint256(currentFixedRateWad) - _sigmaWad), _max_possible_lower_bound);
+            uint256 deltaWad = uint256(currentFixedRateWad) - _sigmaWad;
```

Comment:
deltaWad has to be an int256 rather than a uint256, wdyt @0xZenus ?

Reply:
Definitely.

Reply:
It'd be also good to UTs where `current fixed rate -  sigma` is below 0.001 (the lowest fixed rate supported by our vamm) and also where `current fixed rate +  sigma` is above 1000. 

The more cases for UTs, the better in these scenarios.